avm/ptn/lz/sub-vending/src/modules/Microsoft.Authorization/roleAssignments/resourceGroup/deploy.bicep

```diff
@@ -0,0 +1,89 @@
+targetScope = 'resourceGroup'
```

Comment:
General question @jtracey93,
should pattern modules follow the same folder structure as resource modules? For example, for resource modules (and other ptn modules) we argued that nested templates should be in a `modules` subfoder of the module's root folder. 
However, in this module there is a `/src/` folder in between... A similar question would arise for the scripts as I don't think we have any spec on that as none of our modules ever needed scripts (but only tests).

Reply:
Makes sense to I suppose @AlexanderSehr , can you change @sebassem ?

Reply:
To quote from a private chat, it would look similar to
```
/
/main.bicep
/modules/
/modules/roleAssignmentsSubscription.bicep
/modules/roleAssignmentsMgmtGroup.bicep
/modules/tagsSubscription.bicep
/...
```

Reply:
Follow up @sebassem,
we quickly debated the other special convention of the module, that is the `scripts` scripts folder in the `src` . The agreed convention is to move this into root, so the end result would look like
```
/
/main.bicep
/modules/
/modules/roleAssignmentsSubscription.bicep
/modules/roleAssignmentsMgmtGroup.bicep
/modules/tagsSubscription.bicep
/scripts/
/scripts/myFirstScript.ps1/
/...
```

PS: The templates of the `self` folder should also move into the `modules` folder